R/convertToEGRET.R

```diff
@@ -139,50 +153,54 @@ convertToEGRETDaily <- function(estdat, meta, preds, preds.type = "Conc", qconve
     return(NA)
   }
   
-  stopifnot(preds.type %in% c('Conc', 'Flux'))
-
+  # from https://github.com/USGS-R/EGRET/blob/0a44aa92c8f473ffd67742c866588d45e3e4d8c9/R/estSurfaces.R#L5-L8:
+  # the EGRET surfaces/columns are:
+  #   (1) is the estimated log concentration (yHat), 
+  #   (2) is the estimated standard error (SE), 
+  #   (3) is the estimated concentration (ConcHat). 
+  
+  stopifnot(preds.type == 'Conc') # We need it to be Conc to work with EGRET. No choice.
+  
   daily_df <- flowCorrectionEGRET(flowdat = estdat, 
                                   flow.colname = verify_meta(meta, 'flow'),
                                   date.colname = verify_meta(meta, 'dates'),
                                   qconvert = qconvert)
   
   if(!is.null(preds)) {
     
-    se.pred <- ConcDay <- FluxDay <- '.dplyr.var'
-  
+    fit <- se.pred <- ConcDay <- FluxDay <- '.dplyr.var'
+    
+    # merge daily_df with preds
     daily_df <- daily_df %>% 
       left_join(preds, by=c("dateTime" = "date")) %>% 
-      rename_(.dots = setNames("fit",paste0(preds.type, "Day"))) %>% 
-      rename(SE = se.pred)
-  
-    if(preds.type == "Conc") {
-      daily_df <- mutate(daily_df, FluxDay = ConcDay*daily_df$Q * 86.4)
-    } else {
-      daily_df <- mutate(daily_df, ConcDay = FluxDay/(daily_df$Q * 86.4))
-    }
-      
-    # TO DO: explore issues with bias
-    # ldecicco "In EGRET, it's a little more complicated than just 
-    # log(C) (there's various bias correction things going on)"
-    daily_df <- daily_df %>% mutate(yHat = log(ConcDay))
+      rename(
+        ConcDay = fit,
+        SE = se.pred) %>% # in EGRET, the SE always describes the SE of yHat, so this is wrong. see #172
+      mutate(
+        yHat = log(ConcDay)) # yHat should differ from log(ConcDay) by a model-specific bias correction factor. see #134
+    
+    # fill in the Flux preds based on the Conc preds
+    meta.conv <- updateMetadata(meta, flow='Q', flow.units='cms', load.rate.units='kg d^-1')
```

Comment:
Can we always expect the load rate units to be in kg/d? Where does that conversion happen if the concentration is not in the expected units?